### DISCUSSION TITLE
Scaffolding a la Ruby on Rails

### DISCUSSION THREAD

**Discussion by timmysmalls** [copier-org/copier#400] [Q&A]
Hey guys,

I was wondering if Copier supports scaffolding in a similar way to Ruby on Rails. In the sense that I have an existing project that may or may not have been written with Copier, and now I want to add several related files across the project in one go. Perhaps a migration file, a model file, a controller file, a test file, and for good measure, a few lines in routes.rb. I know I can create single file scaffolds. If I would extend that notion, could scaffolding multiple files across several pre-populated folders be possible? Thank you for your time!

**Comment by yajo** [comment]
I'm not sure for your exact use case, but according to what you say, it should be completely possible.

For example, you can do `copier copy ./template ./destination` even if `./destination` already exists, so you apply the template to the preexisting project, solve any possible conflicts and, from there onwards, you can get updates from the template (if the template supports them).

Another thing you can do is to use a template that doesn't support updates (just for scaffolding) and `copier copy ./tpl-model ./myproject/models/new-model` to scaffold a template into a portion of a preexisting project.

I hope that answers your question.
